src/main.py

```diff
@@ -1,18 +1,243 @@
+BORROWS_FILE = os.path.join(DATA_DIR, 'borrows.json')
+
+def load_data(file):
+    if os.path.exists(file):
+        with open(file, 'r', encoding='utf-8') as f:
+            return json.load(f)
+    return {}
+
+def save_data(file, data):
+    with open(file, 'w', encoding='utf-8') as f:
+        json.dump(data, f, ensure_ascii=False, indent=4)
+
+users = load_data(USERS_FILE)
+books = load_data(BOOKS_FILE)
+borrows = load_data(BORROWS_FILE)
+
+
+
+def login():
+    username = input("Felhasználónév: ")
+    password = input("Jelszó: ")
+    if username in users and users[username]['password'] == password:
+        print(f"Üdvözlünk, {username}!")
+        return {'username': username, **users[username]}
+    print("Hibás felhasználónév vagy jelszó.")
+    return None
+
+def register():
+    username = input("Felhasználónév: ")
+    if username in users:
+        print("Ez a felhasználónév már foglalt.")
+        return None
+    password = input("Jelszó: ")
+    users[username] = {'password': password, 'is_admin': False}
+    save_data(USERS_FILE, users)
+    print("Sikeres regisztráció!")
+    return {'username': username, **users[username]}
+
+def list_books():
+    print("\nKönyvek listája:")
+    for registry_number, book in books.items():
+        print(f"Nyilvántartási szám #{registry_number} Cím: {book['title']}, Író: {book['author']}, Kategória: {book['category']}, Elérhető példányok: {book['copies']}")
+
+def add_book():
+    title = input("Cím: ")
+    author = input("Író: ")
+    registry_number = input("Nyilvántartási szám: ")
+    category = input("Kategória: ")
+    copies = int(input("Példányszám: "))
+    if registry_number in books:
+        print("Ez a nyilvántartási szám már létezik.")
+        return
+    books[registry_number] = {'title': title, 'author': author, 'category': category, 'copies': copies, 'available':True}
+    save_data(BOOKS_FILE, books)
+    print("Könyv hozzáadva!")
+
+
+def borrow_book(user):
+    list_books()
+    registry_number = input("Add meg a kölcsönözni kívánt könyv nyilvántartási számát: ")
+    if registry_number in books and books[registry_number]['copies'] > 0:
+        if registry_number in borrows and user['username'] in [borrow['username'] for borrow in borrows[registry_number]]:
+            print("Ebből a könyvből már kölcsönöztél.")
+            return
+        borrow_date = datetime.now().strftime('%Y-%m-%d')
+        if registry_number not in borrows:
+            borrows[registry_number] = []
+        borrows[registry_number].append({'username': user['username'], 'borrow_date': borrow_date})
+        books[registry_number]['copies'] -= 1
+        if books[registry_number]['copies'] == 0:
+            books[registry_number]['available'] = False 
+        save_data(BOOKS_FILE, books)
+        save_data(BORROWS_FILE, borrows)
+        print("Könyv kölcsönözve!")
     else:
-        name = input("Enter your name: ")
-    print(greeting(name))
+        print("A könyv nem elérhető.")
```

Comment:
Csak egy kis stilisztika, a 2. if-nél jól használtad az early returnt, az 1.-nél is szebb így, else nélkül:
```suggestion
    if registry_number not in books or books[registry_number]['copies'] == 0:
        print("A könyv nem elérhető.")
        return
    if registry_number in borrows and user['username'] in [borrow['username'] for borrow in borrows[registry_number]]:
        print("Ebből a könyvből már kölcsönöztél.")
        return
    borrow_date = datetime.now().strftime('%Y-%m-%d')
    if registry_number not in borrows:
        borrows[registry_number] = []
    borrows[registry_number].append({'username': user['username'], 'borrow_date': borrow_date})
    books[registry_number]['copies'] -= 1
    if books[registry_number]['copies'] == 0:
        books[registry_number]['available'] = False 
    save_data(BOOKS_FILE, books)
    save_data(BORROWS_FILE, borrows)
    print("Könyv kölcsönözve!")
```